bob-access/src/authenticator/basic.rs

```diff
@@ -60,8 +62,22 @@ impl<Storage: UsersStorage> Basic<Storage> {
                 );
         
                 let user = self.users_storage.get_user(&username)?;
-                if user.password() == password {
-                    Ok(user.into())
+                if let Some(usr_password) = user.password() {
+                    if usr_password == password {
+                        Ok(user.into())
+                    } else {
+                        Err(Error::UnauthorizedRequest)
+                    }
+                } else if let Some(usr_hash) = user.hash() {
+                    let hash_str = format!("{}bob", password);
+                    let mut hasher = Sha512::new();
+                    hasher.update(&hash_str.into_bytes());
+                    let hash = hasher.finalize();
```

Comment:
Because in practice there will be a small number of users, it makes sense to store a limited number of hashes inside local cache (small HashMap, which size is limited to 32, without displacement)